components/changelog/ChangelogTrigger.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { gql } from '@apollo/client';
+import { graphql, withApollo } from '@apollo/client/react/hoc';
+import themeGet from '@styled-system/theme-get';
+import { cloneDeep } from 'lodash';
+import styled from 'styled-components';
+
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+import { getFromLocalStorage, LOCAL_STORAGE_KEYS, setLocalStorage } from '../../lib/local-storage';
+import { parseToBoolean } from '../../lib/utils';
+
+import { Flex } from '../Grid';
+import { withNewsAndUpdates } from '../NewsAndUpdatesProvider';
+import { Dropdown } from '../StyledDropdown';
+
+import ChangelogNotificationDropdown from './ChangelogNotificationDropdown';
+
+const CHANGE_LOG_UPDATES_ENABLED = parseToBoolean(process.env.CHANGE_LOG_UPDATES_ENABLED);
+
+const FlameIcon = styled(Flex)`
+  border-radius: 50%;
+  background-color: ${props => themeGet(props.backgroundColor)};
+  background-image: ${props => `url(${props.url})`};
+  background-repeat: no-repeat;
+  background-position: center center;
+  height: 30px;
+  width: 30px;
+  margin-left: 2px;
+`;
+
+const ChangelogTrigger = props => {
+  const { setShowNewsAndUpdates, setChangelogViewDate } = props;
+  let LoggedInUser;
+  if (typeof window !== 'undefined' && localStorage.getItem(LOCAL_STORAGE_KEYS.LOGGED_IN_USER)) {
+    LoggedInUser = JSON.parse(getFromLocalStorage(LOCAL_STORAGE_KEYS.LOGGED_IN_USER)).value;
+  }
+  const LoggedInUserFromCache = props.client.readQuery({ query: loggedInUserQuery })?.LoggedInUser || LoggedInUser;
```

Comment:
We should keep loading the user with `useUser`

Reply:
I don't understand why we load the user from the cache here?

Reply:
Why not use `LoggedInUser` directly?

Reply:
@Betree : It think you are looking at a older diff? I just changed everything to remove using localstorage completely. 😄 

https://github.com/opencollective/opencollective-frontend/pull/6510/commits/50562f7b5ac955319e8b9c3886f75c7c008fc723

Reply:
By "cache", I meant the Apollo cache

Reply:
@Betree : This would work I think with a little caveat. The only problem here is that when the user navigates to a different route, they will see the notification again. The reason I think is the `LoggedInUser` tends to be refreshed from the localstorage although we have updated it in the cache. But on the second reload/refresh everything will be normal again. 

I've added commit; beea21a which illustrates this. 😄 I don't think this is a huge issue; but the user will get the notification two times which I don't think is ideal. 🤔 How about we keep the cache solution I have for now and then when we fix the localStorage issue we can update this to remove the `readCache` call. 🤔 

![test](https://user-images.githubusercontent.com/12435965/123696556-b0125580-d810-11eb-9721-9d756a2a9e57.gif)

Reply:
@Betree : Furthermore, my thinking is that to properly handle this kinda situations specific to LoggedInUser, we need to make the following requests fetch policy different; 

https://github.com/opencollective/opencollective-frontend/blob/c46c2330cfedff21d5adc78016b675e6f7f0b257/lib/withLoggedInUser.js#L55

Note that it's `network-only` and [from the docs](https://www.howtographql.com/react-urql/7-pagination-and-cache-updates/) it seems that; 

>`network-only` always sends a network request to get a query’s result and ignores the cache. 

Thus my understanding is that anything we probably do to with caching related to the `LoggedInUser` has no effect on the actual `LoggedInUser` object obtained using the `UserProvider`. 🤔 🤔

Reply:
> How about we keep the cache solution I have for now and then when we fix the localStorage issue we can update this to remove the readCache call. 

We never load data using `readCache` (outside of `update functions), it's a workaround around Apollo's standard practices, and it does not seem right to me. In the idea of merging this asap, we can live with it, but we should definitely iterate.

I'm going to checkout the PR to try things locally as I'm not sure to understand the root cause completely.

Reply:
@Betree : Yes, agreed that it doesn't seem exactly right to me either, but I couldn't come up with a better alternative yet. We can wait I think a bit more before merging as I don't think there's a rush to do it today itself (correct me if I am wrong). I would rather like a cleaner solution as well. 😄 🐨 

Let me know how things go at your end. I've verified the Apollo cache behaviour using [Apollo Cache Devtools](https://addons.mozilla.org/en-US/firefox/addon/apollo-developer-tools/), in case you find it helpful. Thanks a bunch. 😄

Reply:
Here's the result: https://github.com/opencollective/opencollective-frontend/pull/6552. This version works, even though it has one downside: we're not returning an `id` in the `optimisticResponse` (because we don't know GQLV2's ID for the user from the V1 query). Because this can later cause trouble in the cache, I think we should remove it as the experience of not updating the icon instantaneously is not so bad anyway since it's in the background (behind the modal).

The key is really to load the data from the cache using `fetchPolicy: 'cache-only'`. This approach does not trigger an additional query, yet it does update the component whenever `LoggedInUser` is updated somewhere else (which is not the case with `readQuery`).

```es6
const { data } = useQuery(loggedInUserQuery, { fetchPolicy: 'cache-only' });
```

I'm going to iterate on https://github.com/opencollective/opencollective/issues/4003 to sanitize things here.

Reply:
@Betree : Wow, that's amazing, I've missed this `cache-only` thing completely. I thought the original query where the `LoggedInUser` comes from should be `cache-only`. 😄 

What do you mean by removing it? You mean not integrating your solution? 🤔